DP02_09_Custom_Coadds/DP02_09b_Custom_Coadd_Sources.ipynb

```diff
@@ -345,371 +336,329 @@
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "d56e38a2-2f31-4a00-840f-450c5757c213",
+   "id": "65c4bd4e-469c-47e7-bf3f-2398277941dc",
    "metadata": {},
    "source": [
-    "Run the source detection task.\n",
-    "Send the standard output (a structure) into `result_sd`. \n",
-    "A few lines of information about the task's progress will still display to the screen."
+    "**Option:** to recreate the above `QuantumGraph` visualization yourself, you can uncomment the following cell and run it. Note that running the following optional commands will create two files, `source_detection_qgraph.dot` and `source_detection_qgraph.png`, in your RSP home directory. Note also that running this optional `QuantumGraph` recreation will yield a warning message which can be ignored."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "a39f80cf-f503-4051-9272-4f174b2ac776",
+   "id": "c8c6a0d3-6841-4226-ab29-8c1b0290f2c2",
    "metadata": {},
    "outputs": [],
    "source": [
-    "result_sd = sourceDetectionTask.run(srctab, my_deepCoadd)"
+    "#from lsst.ctrl.mpexec import pipeline2dot\n",
+    "#pipeline2dot(sourceDetectionPipeline, '/home/' + my_username + '/source_detection_qgraph.dot')\n",
+    "\n",
+    "#! dot -Tpng /home/$USER/source_detection_qgraph.dot > /home/$USER/source_detection_qgraph.png"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "7298901b-0c2c-47d9-a34e-2ed764676272",
-   "metadata": {},
-   "source": [
-    "The result returned is a `Struct`, a generalized container. Investigate with `getDict`."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "50d43b41-6cef-4995-9f64-a6fe631ddffc",
+   "id": "ad9057f7-c1fa-47e2-bde5-1d385bbb9cc8",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "for k, v in result_sd.getDict().items():\n",
-    "    print(k, type(v))"
+    "## 2.4. Deploying your source detection pipeline"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "bcec0e96-2392-4a0a-97bb-5053fd0b0b7d",
+   "id": "6a2d3a5c-8daa-435d-9522-c20f0ceffd4f",
    "metadata": {},
    "source": [
-    "Print how many positive peaks were found."
+    "**Important**: You only want to run your source detection pipeline on the specific custom coadd that you've already generated, not all of the DP0.2 `deepCoadd` products. In order to communicate this to the Simple Pipeline Executor framework, you need to define a <a href=\"https://pipelines.lsst.io/modules/lsst.daf.butler/queries.html\">query string</a> that specifies what dataId you wish to generate a source catalog for. As discussed above, the desired tract (patch) to process is 4431 (17), the band is i, and the relevant sky map is that of the DC2 simulation which forms the basis of DP0.2."
```

Comment:
Re. "You only want to run your source detection pipeline on the specific custom coadd that you've already generated, not all of the DP0.2 deepCoadd products" --> In this case, since the input collection only has the one tract/patch/filter could the queryString be empty?

Reply:
Good question, I hadn't actually tried anything like this. However, I just tried a test where I removed the i-band constraint from the query, and I can see that it's now running source detection on the other bands in addition to i-band. So I believe the data query is necessary. I'm not 100% sure, but I think this is due to the "chaining" of the custom coadd collection to the rest of what's in the `dp02` butler repo (specifically the `2.2i/runs/DP0.2` collection). Going to leave this unchanged.